airflow/utils/logging.py

```diff
@@ -211,3 +212,30 @@ def parse_gcs_url(self, gsurl):
             bucket = parsed_url.netloc
             blob = parsed_url.path.strip('/')
             return (bucket, blob)
+
+
+def get_log_filename(**kwargs):
+    """
+    Return full log filename. For example:
+        dag_id/task_id/execution_date/try_number.log
+    """
+    relative_dir = get_log_directory(**kwargs)
+    try_number = kwargs.get('try_number')
+    if try_number is None:
+        raise AirflowException("Try number is required to get log filename")
+
+    # Note try_number index starts from 1.
```

Comment:
That's what I thought as well. Here is why I always add 1 to try number: https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L1312. We can either make everything start from 0, including the name of log file and UI to display the attempt number, or add 1 to everything. Latter make more sense to user and former is better for the codebase.